### ISSUE TITLE
Modify internal PROTO nodes and fields from Supervisor API

### ISSUE THREAD

**Issue by stefaniapedrazzi** [cyberbotics/webots#1923]
Since #1331, it is now possible to modify internal fields of a PROTO node using the Supervisor API.
But this seems fundamentally wrong.
Should not be possible to modify only the PROTO parameters (the same as from the scene tree)?

In issue #1310, it was described the need to access internal PROTO data only in read-only mode but it doesn't seem that the read-only constraint has not been implemented.

**Comment by DavidMansolino** [comment]
This should not be the case, we should only be allowed to read inter fields but not to change them:
https://github.com/cyberbotics/webots/blob/develop/src/lib/Controller/api/supervisor.c#L990
Did you found a case where the `is_proto_internal` flag is not correct?

**Comment by stefaniapedrazzi** [comment]
I have a sample simulation where I can change internal fields.
I will check the `is_proto_internal` flag.

**Comment by stefaniapedrazzi** [comment]
I confirm the `is_proto_internal` flag is not set.
This is because the field is not an internal field of a PROTO node, but a field of a base node defined in a PROTO body.
The current check loops only to the parent node field and doesn't take into account if the ancestor nodes.

**Comment by stefaniapedrazzi** [comment]
Here is a sample node structure where you can change internal fields.

```
PROTO Test [
]
{
Robot {
  children [
    DEF IINTERNAL Solid {
      translation 0 0 0
    }
  ]
}
}
```

controller code:
```
WbNodeRef internal = wb_supervisor_node_get_from_proto_def("INTERNAL");
WbFieldRef internal_field = wb_supervisor_node_get_field(internal, "translation");
const double t[3] = { 0, 0.5, 0 };
wb_supervisor_field_set_sf_vec3f(internal_field, t);
```

**Comment by DavidMansolino** [comment]
Ok, then this is probably a bug :disappointed:

**Comment by stefaniapedrazzi** [comment]
I'm fixing it in #1925.

But checking the new `wb_supervisor_node_get_from_proto_def` and `wb_supervisor_node_get_proto_field` functions, I'm wondering if they should not really work only for PROTO nodes (now they work for every node) and if we should not pass the parent PROTO node reference in order to limit the scope of the search because currently the first DEF found is returned independently if it is in a PROTO body or not.

**Comment by DavidMansolino** [comment]
Why not, but in that case, we should at least provide a function to know if a node is a PROTO or not.

**Comment by omichel** [comment]
Yes, I also believe these functions should work for PROTO nodes onles.
Also `wb_supervisor_node_get_from_proto_def` should take the proto node as a parameter.

**Comment by stefaniapedrazzi** [comment]
Ok. So if we agree I will modify the functions (so that it would also be maybe possible to improve the fix in #1925) and add a new function that returns if a node is a PROTO:
```
wb_supervisor_node_is_proto(WbNodeRef node)
```
